src/drivers/amd/renoir/acp_dmic_dai.c

```diff
@@ -124,7 +139,15 @@ static int acp_dmic_dai_get_hw_params(struct dai *dai,
 		dai_info(dai, "ACP:unsupported samplerate %d", acpdata->dmic_params.pdm_rate);
 		params->rate = ACP_DEFAULT_SAMPLE_RATE;
 	}
-	params->channels = ACP_DEFAULT_NUM_CHANNELS;
+	switch (acpdata->dmic_params.pdm_ch) {
+	case 2:
+	case 4:
+		params->channels = acpdata->dmic_params.pdm_ch;
+		break;
+	default:
+		dai_info(dai, "ACP:unsupported channels %d", acpdata->dmic_params.pdm_ch);
+				 params->channels = ACP_DEFAULT_NUM_CHANNELS;
```

Comment:
misaligned, makes it difficult to notice, it looks like a `dai_info()` continuation

Reply:
changes done